streams/test-utils/src/test/java/org/apache/kafka/streams/TestTopicsTest.java

```diff
@@ -0,0 +1,414 @@
+        //Create Actual Stream Processing pipeline
+        builder.stream(INPUT_TOPIC).to(OUTPUT_TOPIC);
+        final KStream<Long, String> source = builder.stream(INPUT_TOPIC_MAP, Consumed.with(longSerde, stringSerde));
+        final KStream<String, Long> mapped = source.map((key, value) -> new KeyValue<>(value, key));
+        mapped.to(OUTPUT_TOPIC_MAP, Produced.with(stringSerde, longSerde));
+        testDriver = new TopologyTestDriver(builder.build(), config);
+    }
+
+    @After
+    public void tearDown() {
+        try {
+            testDriver.close();
+        } catch (final RuntimeException e) {
+            // https://issues.apache.org/jira/browse/KAFKA-6647 causes exception when executed in Windows, ignoring it
+            // Logged stacktrace cannot be avoided
+            log.warn("Ignoring exception, test failing in Windows due this exception: {}", e.getLocalizedMessage());
+        }
+    }
+
+    @Test
+    public void testValue() {
+        final TestInputTopic<String, String> inputTopic = testDriver.createInputTopic(INPUT_TOPIC, stringSerde.serializer(), stringSerde.serializer());
+        final TestOutputTopic<String, String> outputTopic = testDriver.createOutputTopic(OUTPUT_TOPIC, stringSerde.deserializer(), stringSerde.deserializer());
+        //Feed word "Hello" to inputTopic and no kafka key, timestamp is irrelevant in this case
+        inputTopic.pipeInput("Hello");
+        assertThat(outputTopic.readValue(), equalTo("Hello"));
+        //No more output in topic
+        assertThat(outputTopic.isEmpty(), is(true));
+    }
+
+    @Test
+    public void testValueList() {
+        //Note using here string key serde even other topic is expecting long
+        //Does not affect when key not used
+        final TestInputTopic<String, String> inputTopic = testDriver.createInputTopic(INPUT_TOPIC, stringSerde.serializer(), stringSerde.serializer());
+        final TestOutputTopic<String, String> outputTopic = testDriver.createOutputTopic(OUTPUT_TOPIC, stringSerde.deserializer(), stringSerde.deserializer());
+        final List<String> inputList = Arrays.asList("This", "is", "an", "example");
+        //Feed list of words to inputTopic and no kafka key, timestamp is irrelevant in this case
+        inputTopic.pipeValueList(inputList);
+        final List<String> output = outputTopic.readValuesToList();
+        assertThat(output, hasItems("This", "is", "an", "example"));
+        assertThat(output, is(equalTo(inputList)));
+    }
+
+    @Test
+    public void testKeyValue() {
+        final TestInputTopic<Long, String> inputTopic = testDriver.createInputTopic(INPUT_TOPIC, longSerde.serializer(), stringSerde.serializer());
+        final TestOutputTopic<Long, String> outputTopic = testDriver.createOutputTopic(OUTPUT_TOPIC, longSerde.deserializer(), stringSerde.deserializer());
+        inputTopic.pipeInput(1L, "Hello");
+        assertThat(outputTopic.readKeyValue(), equalTo(new KeyValue<>(1L, "Hello")));
```

Comment:
A minor point, but isn't this supposed to swap the key and value on output?

Reply:
There are two stream. This is using this:
        builder.stream(INPUT_TOPIC).to(OUTPUT_TOPIC);

OUTPUT_TOPIC_MAP is where values are swapped